### PR TITLE
feat(shared): validate response data is JSON object

### PR DESCRIPTION
**Pull request by markscott-ms**
## Description
This pull request improves the robustness of the document loaders by ensuring that only valid JSON objects are accepted as loaded documents. It introduces a shared utility for validating responses and adds comprehensive test coverage for edge cases where the response is not a JSON object.

**Type-confusion safety:** downstream code that assumes the loaded document is an object (e.g. calling `Object.keys(doc)` or accessing `doc.$id`) will no longer receive unexpected behaviour if a remote server returns a string, `null`, or an array. The validation guarantees the shape contract is met before the document is used anywhere in the pipeline.

**Validation improvements:**

* Added a new utility function `assertJsonObject` in `document-loader.ts` to verify that loaded data is a non-null JSON object and not an array; throws a `DocumentLoadError` if validation fails.
* Integrated `assertJsonObject` into both `CalmHubDocumentLoader` and `DirectUrlDocumentLoader` to enforce response validation after fetching documents. [[1]](diffhunk://#diff-eb3d49a4aff3b7b3783ec28c8aa2b7165d29ef57d1bfb5caa38625f01520f1a1R71) [[2]](diffhunk://#diff-9788a5e16c3c66572200702785ded095c237a805d371d49ba84427942e5d1240R124)
* Updated imports in `calmhub-document-loader.ts` and `direct-url-document-loader.ts` to include the new utility. [[1]](diffhunk://#diff-eb3d49a4aff3b7b3783ec28c8aa2b7165d29ef57d1bfb5caa38625f01520f1a1L3-R3) [[2]](diffhunk://#diff-9788a5e16c3c66572200702785ded095c237a805d371d49ba84427942e5d1240L4-R4)

**Test coverage:**

* Added tests in `calmhub-document-loader.spec.ts` and `direct-url-document-loader.spec.ts` to ensure an error is thrown when the response is a string, null, or an array, improving reliability and correctness. [[1]](diffhunk://#diff-4873ba57b2429a3fe359bec455d110ae287fc89f4d933c2634ddf9f3d4e867deR56-R73) [[2]](diffhunk://#diff-d6194dcdff95b132094fa0543dff241a6e13e6fbba872461b21f4dcf89f19703R153-R170)

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🎨 Code style/formatting changes
- [ ] ♻️ Refactoring (no functional changes)
- [ ] ⚡ Performance improvements
- [ ] ✅ Test additions or updates
- [ ] 🔧 Chore (maintenance, dependencies, CI, etc.)

## Affected Components

- [X] CLI (`cli/`)
- [ ] Schema (`calm/`)
- [ ] CALM AI (`calm-ai/`)
- [ ] CALM Hub (`calm-hub/`)
- [ ] CALM Hub UI (`calm-hub-ui/`)
- [ ] CALM Server (`calm-server/`)
- [ ] CALM Widgets (`calm-widgets/`)
- [ ] Documentation (`docs/`)
- [X] Shared (`shared/`)
- [ ] VS Code Extension (`calm-plugins/vscode/`)
- [ ] Dependencies
- [ ] CI/CD

## Testing

- [X] I have tested my changes locally
- [X] I have added/updated unit tests
- [X] All existing tests pass

## Checklist
- [X] My commits follow the [conventional commit format](https://www.conventionalcommits.org/)
- [ ] I have updated documentation if necessary
- [X] I have added tests for my changes (if applicable)
- [X] My changes follow the project's coding standards

### CODE REVIEW

File: shared/src/document-loader/document-loader.ts [hunk 1]
```diff
@@ -64,6 +64,15 @@ export function buildDocumentLoader(docLoaderOpts: DocumentLoaderOptions): Docum
     return new MultiStrategyDocumentLoader(loaders, debug);
 }
 
+export function assertJsonObject(data: unknown, source: string): asserts data is object {
+    if (typeof data !== 'object' || data === null || Array.isArray(data)) {
+        throw new DocumentLoadError({
+            name: 'UNKNOWN',
+            message: `Expected a JSON object from ${source} but received: ${typeof data}`
```
[the remark this answers is not in the document]
Comment (markscott-ms) [reply shared/src/document-loader/document-loader.ts]: Good catch my AI friend.

**Comment by rocketstack-matt** [comment]
Reviewed — overall LGTM as defense-in-depth.

One thing worth surfacing: the new validation errors from `assertJsonObject` will currently be masked by `MultiStrategyDocumentLoader`'s fall-through behaviour — e.g. a `calm:/...` URL that returns a non-object body throws the clear "Expected a JSON object…" error from `CalmHubDocumentLoader`, but the multi-strategy loader catches it, falls through to `DirectUrlDocumentLoader`, and the user ends up seeing "Unsupported URL protocol 'calm:'" instead.

That's a pre-existing design issue in `MultiStrategyDocumentLoader`, not something this PR introduces — but it does make the symptom more visible. I've raised it as a separate bug: #2440. Not a blocker for this PR.

**Review by rocketstack-matt — CHANGES_REQUESTED** [review]
Requesting a small change on the new tests:

The six new tests in `calmhub-document-loader.spec.ts` (lines 57–73) and `direct-url-document-loader.spec.ts` (lines 156–172) only assert on the error message string. The existing test pattern in this file (e.g. `direct-url-document-loader.spec.ts:148` — *"throws DocumentLoadError for disallowed host"*) tightens the contract with `.rejects.toBeInstanceOf(DocumentLoadError)`.

Without that, a future refactor that throws a plain `Error` carrying the same `"Expected a JSON object"` text would still pass — the type contract isn't pinned down. Could you add a `toBeInstanceOf(DocumentLoadError)` assertion to at least one new test per loader (or all six)? That keeps behaviour aligned with the surrounding suite.

**Comment by rocketstack-matt** [comment]
One note on the security framing in the PR description: the OWASP A03 / A08 classification is reasonable as defense-in-depth, but I'd be careful claiming this prevents prototype pollution specifically. To meaningfully defend against that, the validation would also need to reject objects whose own keys include `__proto__`, `constructor`, or `prototype` (recursively, not just at the top level).

What this PR actually buys is **type-confusion safety** — downstream code that calls e.g. `Object.keys(doc)` or `doc.$id` on the loaded document won't crash on a non-object response body, and won't accidentally iterate array indices or string characters. That's a real and worthwhile guarantee, just a more modest one than "prototype pollution mitigation". Worth tightening the wording in the description (or a follow-up PR) so the threat model is accurate.

**Comment by markscott-ms** [comment]
Comments addressed @rocketstack-matt , good for re-review. Will address #2440 separately

**Review by rocketstack-matt — APPROVED** [review]
Thanks @markscott-ms — the new `toBeInstanceOf(DocumentLoadError)` assertions in `bfe8cb81` address the test-type-pinning concern cleanly. The `const promise = …; await expect(promise).rejects.toBeInstanceOf(...); await expect(promise).rejects.toThrow(...)` pattern is a nice tight check on both type and message without re-invoking the loader. All 30 tests pass locally.

Approving. The remaining items (multi-strategy masking — tracked in #2440; PR description wording around prototype pollution vs type confusion — non-blocking) can be picked up separately.
